.github/actions/codesigner-SSL/action.yml

```diff
@@ -0,0 +1,91 @@
+name: "codesigner"
+description: "GitHub Action for CodeSigner by SSL.com"
+
+inputs:
+  file_path:
+    description: "Path of the code object to be signed. Supported file types: acm, ax, bin, cab, cpl, dll, drv, efi, exe, mui, ocx, scr, sys, tsp, msi, ps1, ps1xml, js, vbs, wsf, jar"
+    required: true
+  output_dir_path:
+    description: "Directory where the signed code object will be written (leave empty if override is set to true)."
+    required: false
+  malware_block:
+    description: "Scans your file for possible malware to prevent code compromise and prevents signing code if malware is detected."
+    required: false
+    default: "false"
+  override:
+    description: "Override the input file after signing if this argument is set to true and no output_dir_path argument was passed."
+    required: false
+    default: "true"
+  clean_logs:
+    description: "Clean log files after code signing operations."
+    required: false
+    default: "false"
+  jvm_max_memory:
+    description: "Maximum JVM heap size."
+    required: false
+    default: "1024M"
+  signing_method:
+    description: "Code signing method. Default is v1. Supported values: v1, v2."
+    required: false
+    default: "v1"
+  ES_USERNAME:
+    description: "SSL.com account username."
+    required: true
+  ES_PASSWORD:
+    description: "SSL.com account password."
+    required: true
+  ES_CREDENTIAL_ID:
+    description: "Credential ID for signing certificate."
+    required: true
+  ES_TOTP_SECRET:
+    description: "OAuth TOTP Secret."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+      - name: Sign Artifact with CodeSignTool
+        uses: sslcom/esigner-codesign@v1.3.1
+        with:
+          # CodeSignTool Commands:
+          # - get_credential_ids: Output the list of eSigner credential IDs associated with a particular user.
+          # - credential_info: Output key and certificate information related to a credential ID.
+          # - sign: Sign and timestamp code object.
+          # - batch_sign: Sign and timestamp multiple code objects with one OTP.
+          # - hash: Pre-compute hash(es) for later use with batch_hash_sign command.
+          # - batch_sign_hash: Sign hash(es) pre-computed with hash command.
+          command: sign

```

Comment:
I also tried the `batch_sign` mode for signing of several files at once, but it seems to be buggy and unstable.
* overwriting of files does not work
* trying to sign the files in the root folder produces weird logging errors
* it tries to sign all files in a folder. If a folder contains a file type which is not supported (e.g. text file) - workflow fails

Reply:
Meaning that if we want to sign several files: we will need to define several steps in the GH-Workflow - 1 step per file to be signed